libs/test-common/src/main/java/org/codice/ddf/test/common/features/FeatureUtilities.java

```diff
@@ -228,13 +207,12 @@ private static void printInactiveBundles(
     }
   }
 
-  public static void printInactiveBundles() {
-    printInactiveBundles(LOGGER::error, LOGGER::error);
-  }
-
-  public static void waitForRequiredBundles(String symbolicNamePrefix) throws InterruptedException {
+  public void waitForRequiredBundles(String symbolicNamePrefix) throws InterruptedException {
     boolean ready = false;
-    BundleService bundleService = getService(BundleService.class);
+
+    if (bundleService == null) {
```

Comment:
❓ Why aren't we using blueprint to inject the bundle service? If we can use blueprint can we remove all these methods related to getting services from the bundle context?

Reply:
I've retested this and investigated some more. This is the exception that's being thrown. It appears that the interaction between pax-exam, blueprint, and Karaf's featureService is breaking down, and the service reference for `BundleService` is not being updated when bundle refreshes are happening even though I've verified the `Import-Service` header for `BundleService` is on this bundle. It's an intermittent issue that depends on timing, though it seems to happen most often with the first feature install in the itests. At this point, I don't think it's worth trying to change.
```
org.osgi.service.blueprint.container.ServiceUnavailableException: The Blueprint container is being or has been destroyed: (objectClass=org.apache.karaf.bundle.core.BundleService)
        at org.apache.aries.blueprint.container.ReferenceRecipe.getService(ReferenceRecipe.java:276)
        at org.apache.aries.blueprint.container.ReferenceRecipe.access$000(ReferenceRecipe.java:57)
        at org.apache.aries.blueprint.container.ReferenceRecipe$ServiceDispatcher.call(ReferenceRecipe.java:341)
        at Proxybea323ac_6212_49bd_a326_9f554304889a.getInfo(Unknown Source)
        at org.codice.ddf.test.common.features.FeatureUtilities.waitForRequiredBundles(FeatureUtilities.java:266)
        at org.codice.ddf.test.common.features.FeatureUtilities.installFeature(FeatureUtilities.java:200)
        at org.codice.ddf.test.common.features.FeatureUtilities.installAndUninstallFeature(FeatureUtilities.java:168)
        at org.codice.ddf.features.camel.test.ITCamelFeatures.installAndUninstallFeature(ITCamelFeatures.java:80)
```